apex/apexLexer.g4

```diff
@@ -0,0 +1,497 @@
+    |   Digits FloatTypeSuffix
+    ;
+
+fragment
+ExponentPart
+    :   ExponentIndicator SignedInteger
+    ;
+
+fragment
+ExponentIndicator
+    :   [eE]
+    ;
+
+fragment
+SignedInteger
+    :   Sign? Digits
+    ;
+
+fragment
+Sign
+    :   [+-]
+    ;
+
+fragment
+FloatTypeSuffix
+    :   [fFdD]
+    ;
+
+fragment
+HexadecimalFloatingPointLiteral
+    :   HexSignificand BinaryExponent FloatTypeSuffix?
+    ;
+
+fragment
+HexSignificand
+    :   HexNumeral '.'?
+    |   '0' [xX] HexDigits? '.' HexDigits
+    ;
+
+fragment
+BinaryExponent
+    :   BinaryExponentIndicator SignedInteger
+    ;
+
+fragment
+BinaryExponentIndicator
+    :   [pP]
+    ;
+
+// ?3.10.3 Boolean Literals
+
+BooleanLiteral
+    :   'true'
+    |   'false'
+    ;
+
+// ?3.10.4 Character Literals
+
+CharacterLiteral
+    :   QUOTE SingleCharacter QUOTE
+    |   QUOTE EscapeSequence QUOTE
+    ;
+
+fragment
+SingleCharacter
+    :   ~['\\]
+    ;
+
+// ?3.10.5 String Literals
+
+StringLiteral
+    :   QUOTE StringCharacters? QUOTE
+    ;
+
+fragment
+StringCharacters
+    :   StringCharacter+
+    ;
+
+fragment
+StringCharacter
+    :   ~['\\]
+    |   EscapeSequence
+    ;
+
+// ?3.10.6 Escape Sequences for Character and String Literals
+
+fragment
+EscapeSequence
+    :   '\\' [btnfr"'\\]
+    |   OctalEscape
+    |   UnicodeEscape
+    ;
+
+fragment
+OctalEscape
+    :   '\\' OctalDigit
+    |   '\\' OctalDigit OctalDigit
+    |   '\\' ZeroToThree OctalDigit OctalDigit
+    ;
+
+fragment
+UnicodeEscape
+    :   '\\' 'u' HexDigit HexDigit HexDigit HexDigit
+    ;
+
+fragment
+ZeroToThree
+    :   [0-3]
+    ;
+
+// ?3.10.7 The Null Literal
+
+NullLiteral :   N U L L;
+
+
+// ?3.11 Separators
+
+LPAREN          : '(';
+RPAREN          : ')';
+LBRACE          : '{';
+RBRACE          : '}';
+LBRACK          : '[';
+RBRACK          : ']';
+SEMI            : ';';
+COMMA           : ',';
+DOT             : '.';
+
+// ?3.12 Operators
+
+ASSIGN          : '=';
+GT              : '>';
+LT              : '<';
+BANG            : '!';
+TILDE           : '~';
+QUESTION        : '?';
+COLON           : ':';
+EQUAL           : '==';
+LE              : '<=';
+GE              : '>=';
+NOTEQUAL        : '!=';
+NOTEQUALGL      : '<>';
+AND             : '&&';
+OR              : '||';
+INC             : '++';
+DEC             : '--';
+ADD             : '+';
+SUB             : '-';
+MUL             : '*';
+DIV             : '/';
+BITAND          : '&';
+BITOR           : '|';
+CARET           : '^';
+MOD             : '%';
+
+ADD_ASSIGN      : '+=';
+SUB_ASSIGN      : '-=';
+MUL_ASSIGN      : '*=';
+DIV_ASSIGN      : '/=';
+AND_ASSIGN      : '&=';
+OR_ASSIGN       : '|=';
+XOR_ASSIGN      : '^=';
+MOD_ASSIGN      : '%=';
+LSHIFT_ASSIGN   : '<<=';
+RSHIFT_ASSIGN   : '>>=';
+URSHIFT_ASSIGN  : '>>>=';
+LAMBDA_LIKE     : '=>';
+
+
+// ?3.8 Identifiers (must appear after all keywords in the grammar)
+
+Identifier
+    :   JavaLetter JavaLetterOrDigit*
+    ;
+
+fragment
+JavaLetter
```

Comment:
https://github.com/antlr/grammars-v4/blob/e53d7a1228505bfc80d8637808ef60e7eea92cc2/java/java8/Java8Lexer.g4#L469

We can make it target independent.

Reply:
Goodness. This code--and Java9--both need to be cleaned up, pronto! In addition, the commented out code in Java8Lexer.g4 needs to be just removed. At first, I was going to ask you how this helps because I was seeing the semantic action code above the [IdentifierStart](https://github.com/antlr/grammars-v4/blob/e53d7a1228505bfc80d8637808ef60e7eea92cc2/java/java8/Java8Lexer.g4#L469) rule. But, then I noticed it was commented out. The commented-out code should all be removed. Really, let's use the power of version control if one needs to see an older version of the grammar.

I'll change this issue and broaden it to include Java8, Java9, and Apex.